package Cmd;

enum EVarType
{
  EVARTYPE_MIN = 0;
  EVARTYPE_QUEST_WANTED = 1;
  EVARTYPE_QUEST_WANTED_RESET = 2;
  EVARTYPE_SHOP = 3;
  EVARTYPE_TOWER = 4;
  EVARTYPE_SEAL = 5;
  EVARTYPE_QUEST_REWARD = 6;            // 任务运营活动
  EVARTYPE_ANTI_ADDICT_DAILY = 7;
  EVARTYPE_LABORATORY = 8;
  EVARTYPE_GUILD_EXCHANGECHAIR = 9;
  EVARTYPE_GUILD_CONTRIBUTION = 10;
  EVARTYPE_USEITEM_DAY = 11;
  EVARTYPE_SINGLE_DOG = 12;             // 废弃
  EVARTYPE_QUEST_DAILY_1 = 13;
  EVARTYPE_QUEST_DAILY_3 = 14;
  EVARTYPE_QUEST_DAILY_7 = 15;
  EVARTYPE_QUEST_DAILY_MAP = 16;
  EVARTYPE_GUILD_ASSET = 17;
  EVARTYPE_CAMERA_SUMMON_DAILY = 18;
  EVARTYPE_ACTIVITY_QUEST = 19;
  EVARTYPE_GUILD_DONATE = 20;
  EVARTYPE_OPERATE_REWARD = 21;        // 临时功能
  EVARTYPE_FIRST_EXCHANGEZONE = 22;
  EVARTYPE_FRIENDSHIP_FRIEND = 23;
  EVARTYPE_FRIENDSHIP_GUILD = 24;
  EVARTYPE_TOWER_MONSTER = 25;
  EVARTYPE_USEITEM_WEEK = 26;
  EVARTYPE_USER_CREDIT = 27;
  EVARTYPE_DAY_ONLINE_FIRST = 28;
  EVARTYPE_ACTIVITY_REWARD = 29;
  EVARTYPE_ACTIVITY_VALENTINE = 30;
  EVARTYPE_CHAT = 31;
  EVARTYPE_AUGURY_REWARD = 32;
  EVARTYPE_SELL_WARNING_LAST = 33;
  EVARTYPE_SELL_WARNING_CUR = 34;
  EVARTYPE_GUILD_RAID = 35;    // 公会随机副本
  EVARTYPE_GUILD_RAID_BAN = 36;
  EVARTYPE_LABORATORY_EXTASKREWARD = 37;
  EVARTYPE_GUILD_QUEST = 38;
  EVARTYPE_CHILD_QUEST = 39;
  EVARTYPE_GUILD_MAXASSET = 40;
  EVARTYPE_LABORATORY_POINT = 41;
  EVARTYPE_GETITEM_DAY = 42;
  EVARTYPE_GETITEM_WEEK = 43;
  EVARTYPE_NEWAUGURY_REWARD = 44;
  EVARTYPE_NEWAUGURY_EXTRACOUNT = 45;
  EVARTYPE_AUGURY_CELEBRATION_DAY = 46;
  EVARTYPE_CELEBRATION_ONE = 47;
  EVARTYPE_CELEBRATION_TWO = 48;
  EVARTYPE_CELEBRATION_THREE = 49;
  EVARTYPE_PVPCOIN_DAY  = 50;
  EVARTYPE_PVPCOIN_WEEK  = 51;
  EVARTYPE_ACHIEVE_CHAT_WORLD = 52;
  EVARTYPE_ACHIEVE_CHAT_GUILD = 53;
  EVARTYPE_QUEST_DAILY_RESET = 54;
  EVARTYPE_SCENERY_UPLOAD = 55;
  EVARTYPE_GUILD_QUEST_EXTRAREWARD = 56;
  EVARTYPE_GUILD_DONATE_EXTRAREWARD = 57;
  EVARTYPE_ACHIEVE_CAT = 58;
  EVARTYPE_LOVEAUGURY_EXTRARCOUNT = 59;
  EVARTYPE_USERPET_TOUCH = 60;
  EVARTYPE_USERPET_FEED = 61;
  EVARTYPE_USERPET_GIFT = 62;
  EVARTYPE_PETADVENTURE_RESET = 63;
  EVARTYPE_SHOP_RANDOM = 64;
  EVARTYPE_ACC_QUEST = 65;
  EVARTYPE_PETADVENTURE_LIST = 66;
  EVARTYPE_LOTTERY_CNT_EQUIP = 67;
  EVARTYPE_LOTTERY_CNT_CARD = 68;
  EVARTYPE_ACCDAILY_QUEST = 69;
  EVARTYPE_TUTOR_TASK_DAY = 70;
  EVARTYPE_TUTOR_TASK_WEEK = 71;
  EVARTYPE_SHOP_RANDOM_BY_LV = 72;
  EVARTYPE_INACTIVE_USER_SEND_COUNT = 73;     //废弃
  EVARTYPE_INACTIVE_USER_LAST_SEND_DAY = 74;  //废弃
  EVARTYPE_QUEST_WANTED_WEEK = 75;            //废弃
  EVARTYPE_TUTOR_TASK_REWARD = 76;
  EVARTYPE_DOUBLE_LAB = 77;
  EVARTYPE_GUILD_PHOTO_LOAD = 78;
  EVARTYPE_ACC_CHOICE_QUEST = 79;
  EVARTYPE_GUILD_BUILDING_SUBMIT_DAY = 85;
  EVARTYPE_DAILY_MAPRAND = 86;
  EVARTYPE_DEPOSIT_END_NTF = 87;
  EVARTYPE_ACTIVITY_EVENT_REWARD = 88;
  EVARTYPE_LABORATORY_COUNT = 89;
  EVARTYPE_GUILD_DONATE_DAY = 90;
  EVARTYPE_GUILD_CHALLENGE_WEEK = 91;
  EVARTYPE_CHRISTMAS_CAKE = 92;
  EVARTYPE_SHOP_WEEK = 93;
  EVARTYPE_TOWER_RESETTIME = 95;
  EVARTYPE_QUEST_SIGN = 96;
  EVARTYPE_ARTIFACT_DISTRIBUTE_DAY = 97;
  EVARTYPE_FIRST_SHARE = 98;

  EVARTYPE_EXTRARWD_WANTEDQUEST = 99;
  EVARTYPE_EXTRARWD_DAILYMONSTER = 100;
  EVARTYPE_EXTRARWD_SEAL = 101;
  EVARTYPE_EXTRARWD_LABORATORY = 102;
  EVARTYPE_EXTRARWD_ENDLESS = 103;
  EVARTYPE_EXTRARWD_GUILD_QUEST = 104;
  EVARTYPE_EXTRARWD_GUILD_DONATE = 105;

  EVARTYPE_DOUBLERWD_WANTEDQUEST = 106;
  EVARTYPE_DOUBLERWD_DAILYMONSTER = 107;
  EVARTYPE_DOUBLERWD_SEAL = 108;
  EVARTYPE_DOUBLERWD_LABORATORY = 109;
  EVARTYPE_DOUBLERWD_ENDLESS = 110;

  EVARTYPE_SHOP_MONTH = 111;
  EVARTYPE_GUILD_TREASURE_COUNT = 112;
  EVARTYPE_BCOIN_TREASURE_COUNT = 113;
  EVARTYPE_DAY_LOTTERY_BUY_GIVE_CNT = 114;
  EVARTYPE_DAY_LOTTERY_CNT_HEAD = 115;
  EVARTYPE_DAY_LOTTERY_CNT_EQUIP = 116;
  EVARTYPE_DAY_LOTTERY_CNT_CARD = 117;
  EVARTYPE_DAY_LOTTERY_CNT_MAGIC = 118;
  EVARTYPE_QUEST_WEDDINGDAILY = 120;
  EVARTYPE_RECOMMEND_DAY = 123;
  EVARTYPE_RECOMMEND_WEEK = 124;

  EVARTYPE_PVECARD_DIFFICULTY_1 = 132;
  EVARTYPE_PVECARD_DIFFICULTY_2 = 133;
  EVARTYPE_PVECARD_DIFFICULTY_3 = 134;
  EVARTYPE_MVPREWARDNUM = 135;
  EVARTYPE_MINIREWARDNUM  = 136;
  EVARTYPE_ALTMAN_REWARD = 137;
  EVARTYPE_ALTMAN_KILL = 138;
  EVARTYPE_EXTRARWD_PVECARD  = 139;
  EVARTYPE_DOUBLERWD_PVECARD = 140;
  EVARTYPE_GUILD_MEMBER_WEEKBCOIN = 142;
  EVARTYPE_EXCHANGECARD_DRAWMAX = 143;
  EVARTYPE_DAY_GET_ZENY_COUNT = 144;
  EVARTYPE_DEAD_COIN = 145;
  EVARTYPE_DEAD_QUEST = 146;
  EVARTYPE_TUTOR_EXTRABATTLETIME = 148;
  EVARTYPE_DEADBOSS_COUNT_MVP = 149;
  EVARTYPE_DEADBOSS_COUNT_PVECARD = 150;
  EVARTYPE_DEADBOSS_COUNT_TOWER = 151;
  EVARTYPE_DEADBOSS_COUNT_GUILD = 152;
  EVARTYPE_TEAMPWS_COUNT = 153;
  EVARTYPE_DEADBOSS_COUNT_PVECARD2 = 158;
  EVARTYPE_DEADBOSS_COUNT_PVECARD3 = 159;
  EVARTYPE_DEADBOSS_COUNT_PVECARD4 = 160;
  EVARTYPE_MAX = 161;
}

enum EVarTimeType
{
  EVARTIMETYPE_MIN = 0;
  EVARTIMETYPE_DAY = 1;
  EVARTIMETYPE_WEEK = 2;
  EVARTIMETYPE_MONTH = 3;
  EVARTIMETYPE_ALWAYS = 4;
  EVARTIMETYPE_MAX = 5;
}

message Var
{
  optional EVarType type = 1 [ default = EVARTYPE_MIN ];
  optional uint32 value = 2 [ default = 0 ];
  optional uint32 time = 3 [ default = 0 ];
}

message StatVar
{ 
  optional string key = 1;
  optional uint32 value = 2 [ default = 0 ];
  optional uint32 time = 3 [ default = 0 ];
}

enum EAccVarType
{
  EACCVARTYPE_MIN = 0;
  EACCVARTYPE_LOTTERY_CNT_EQUIP = 1;
  EACCVARTYPE_LOTTERY_CNT_CARD = 2;
  EVARTYPE_SHOP_RANDOM_BY_ACCLV = 3;
  EACCVARTYPE_DAILY_QUEST = 4;
  EACCVARTYPE_INACTIVE_USER_SEND_COUNT = 5;
  EACCVARTYPE_INACTIVE_USER_LAST_SEND_DAY = 6;
  EACCVARTYPE_QUEST_WANTED_WEEK = 7;
  EACCVARTYPE_DOUBLE_LAB = 8;
  EACCVARTYPE_ACTIVITY_EVENT_REWARD = 9;

  EACCVARTYPE_POLLY_FIRST = 10;
  EACCVARTYPE_POLLY_DAY_SCORE = 11;
  EACCVARTYPE_SHOP_RANDOM_ACC = 13;
  EACCVARTYPE_SHOP_GOT = 14;
  EACCVARTYPE_QUEST_DAILY_RESET = 15;
  EACCVARTYPE_JOY = 16;
  EACCVARTYPE_QUEST_DAILY_1 = 17;
  EACCVARTYPE_QUEST_DAILY_3 = 18;
  EACCVARTYPE_QUEST_DAILY_7 = 19;

  EACCVARTYPE_EXTRARWD_WANTEDQUEST = 20;
  EACCVARTYPE_EXTRARWD_DAILYMONSTER = 21;
  EACCVARTYPE_EXTRARWD_SEAL = 22;
  EACCVARTYPE_EXTRARWD_LABORATORY = 23;
  EACCVARTYPE_EXTRARWD_ENDLESS = 24;
  EACCVARTYPE_EXTRARWD_GUILD_QUEST = 25;
  EACCVARTYPE_EXTRARWD_GUILD_DONATE = 26;

  EACCVARTYPE_DOUBLERWD_WANTEDQUEST = 27;
  EACCVARTYPE_DOUBLERWD_DAILYMONSTER = 28;
  EACCVARTYPE_DOUBLERWD_SEAL = 29;
  EACCVARTYPE_DOUBLERWD_LABORATORY = 30;
  EACCVARTYPE_DOUBLERWD_ENDLESS = 31;

  EACCVARTYPE_SHOP_GOT_WEEK = 32;
  EACCVARTYPE_SHOP_GOT_MONTH = 33;

  EACCVARTYPE_PETWORK_EXCHANGE = 34;
  EACCVARTYPE_PETWORK_REWARD = 35;
  EACCVARTYPE_OPERATE_REWARD = 36;
  EACCVARTYPE_FAVORABILITY = 37;
  EACCVARTYPE_FAVORABILITY_STATUS = 38;


  EACCVARTYPE_EXTRARWD_PVECARD = 39;
  EACCVARTYPE_DOUBLERWD_PVECARD = 40;

  EACCVARTYPE_STAY_FAVORABILITY = 41;
  EACCVARTYPE_MAX = 42;
}

message AccVar
{ 
  optional EAccVarType type = 1 [ default = EACCVARTYPE_MIN ];
  optional uint32 value = 2 [ default = 0 ];
  optional uint32 time = 3 [ default = 0 ];
}
